### DISCUSSION TITLE
incomplete docs?

### DISCUSSION THREAD

**Discussion by hi-reeve** [DamianOsipiuk/vue-query#87] [General]
hello i really want to try this library, and i just reading the whole docs but i cannot find anything related to the mutation. 
does mutation possible with this library?
or did i just miss the mutation section somewhere in the docs?

**Comment by wobsoriano** [comment]
It's the same as the react version https://react-query.tanstack.com/reference/useMutation

```ts
const addPost = useMutation((body) =>
  axios.post('/api/posts', body)
);

addPost.mutate({ ... })
```

**Comment by DamianOsipiuk** [comment]
@zynth17 As @wobsoriano mentioned most of the react-query docs apply. The API and options are exactly the same.
The only thing you need to worry about is adjusting it to the vue reactivity system (so accessing returned values via `.value`).
Most of the react-query docs would be just a copy-paste and i really need to do it some day 😄

**Comment by hi-reeve** [comment]
@wobsoriano @DamianOsipiuk thankyou for the explanation! 😁

> @zynth17 As @wobsoriano mentioned most of the react-query docs apply. The API and options are exactly the same.
> 
> The only thing you need to worry about is adjusting it to the vue reactivity system (so accessing returned values via `.value`).
> 
> Most of the react-query docs would be just a copy-paste and i really need to do it some day 😄 

you can try use unref? or wait for the new script setup ref release.

anyway thankyou for the fast reply. much appreciate it.

**Comment by DamianOsipiuk** [comment]
I've added a paragraph about react-query docs to the readme.

As for refs, yes it's kind of a pain currently. I think the best way would be to use the ref syntax sugar that is in the pipelines of vue - https://github.com/vuejs/rfcs/discussions/369

Also you can do something like `const query = reactive(useQuery(...))` - but then you can't destructure it or you will loose reactivity.

**Comment by hi-reeve** [comment]
> I've added a paragraph about react-query docs to the readme.
> 
> 
> 
> As for refs, yes it's kind of a pain currently. I think the best way would be to use the ref syntax sugar that is in the pipelines of vue - https://github.com/vuejs/rfcs/discussions/369
> 
> 
> 
> Also you can do something like `const query = reactive(useQuery(...))` - but then you can't destructure it or you will loose reactivity.

thankyou!
will take note on that one.
